### PR TITLE
MAGECLOUD-1829: Move Docker configuration from ECE-Tools to Hub

### PR DESCRIPTION
**Pull request by shiftedreality**
https://magento2.atlassian.net/browse/MAGECLOUD-1829

### Fixed Issues (if relevant)

- No Docker build configuration stored in ECE-Tools anymore
- No local integration environment setup now (No PHP, Mysql, etc)
- Static and unit tests were merged together to save Travis CI concurrent jobs
- All jobs, including static, are run in isolated Docker containers o Added possibility to use custom PHPUnit configs
- 3x speed improvement (33min with warm-upped cache for longest build)
- Integration 7.0 test run only on 7.0-related tests (~7 mins)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request was approved by architect
 - [ ] Pull request was approved by QA member
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by JacobBrownAustin — COMMENTED** [review]
Overall, I like that we are separating the docker from the ece-tools.  I think that is great.  There were a few questions I asked, and a few places where potential shell injections need to be better escaped and/or validated.

**Review by billygilbert — CHANGES_REQUESTED** [review]
I think overall it looks fine but the pathing seems to be hard to follow. I think we should rename /var/www/sandbox and be a bit more clean on what each of the /var/www/* directories do.

I also think we should make sure there's a readme.md somewhere (probably docker hub).

**Review by billygilbert — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]
